File: AI_interviewer/client/src/api/Interview_position.ts
import request from '../utils/request'

export interface Interviewer {
  id: number
  name: string
  title?: string
  description?: string
  avatar?: string
  position_name?: string
}

export interface Position {
  id: number 
  position_name: string
  description?: string
  interviewers: Interviewer[]
} 


export function getPosition() {
    return request<any, Position[]>({
    url: '/api/interviewee/get_position',
    method: 'get',
  })
}

File: AI_interviewer/backstage/app/schemas/Interview_position.py
from pydantic import BaseModel
from typing import Optional, List

class Position(BaseModel):
    id: int
    position_name: str
    description: Optional[str] = None  # ✅ 允许 None

    class Config:
        orm_mode = True  # ✅ 启用 ORM 模式

class InterviewerInfo(BaseModel):
    id: int
    name: str
    title: Optional[str] = None
    description: Optional[str] = None
    avatar: Optional[str] = None

    class Config:
        orm_mode = True

class PositionWithInterviewers(BaseModel):
    id: int
    position_name: str
    description: Optional[str] = None
    interviewers: List[InterviewerInfo] = []

    class Config:
        orm_mode = True
